packages/mask/src/plugins/Avatar/Services/gun.ts

```diff
@@ -21,19 +21,25 @@ export async function getUserAddress(userId: string) {
 }
 
 export async function setUserAddress(userId: string, address: string) {
-    // delete userId
-    await NFTAvatarGUN
-        //@ts-expect-error
-        .get(userId)
-        //@ts-expect-error
-        .put(null).then!()
-
-    // save userId
-    await NFTAvatarGUN
-        // @ts-expect-error
-        .get(userId)
-        // @ts-expect-error
-        .put(address).then!()
+    try {
+        // delete userId
+        await NFTAvatarGUN
+            //@ts-expect-error
+            .get(userId)
+            //@ts-expect-error
+            .put(null).then!()
+
+        // save userId
+        await NFTAvatarGUN
+            // @ts-expect-error
+            .get(userId)
+            // @ts-expect-error
+            .put(address).then!()
+
+        const address = await getUserAddress(userId)
```

Comment:
Please move this statement to the final block. If the address is not the same with the original one then you can throw the error. Otherwise, you may throw an error.